memento.root/usr/share/clearwater/cassandra-schemas/memento.sh

```diff
@@ -1,8 +1,119 @@
-#! /bin/bash
-if [[ ! -e /var/lib/cassandra/data/memento ]];
-then
-    echo "CREATE KEYSPACE memento WITH strategy_class='org.apache.cassandra.locator.SimpleStrategy' AND strategy_options:replication_factor=2;
-          USE memento;
-          CREATE TABLE call_lists (impu text PRIMARY KEY, dummy text) WITH read_repair_chance = 1.0;
-    " | cqlsh -2
+#!/bin/bash
+
+keyspace=$(basename $0|sed -e 's#^\(.*\)[.]sh$#\1#')
+
+. /etc/clearwater/config
+if [ ! -z $signaling_namespace ]; then
+    if [ $EUID -ne 0 ]; then
+        echo "When using multiple networks, schema creation must be run as root"
+        exit 2
+    fi
+    namespace_prefix="ip netns exec $signaling_namespace"
+fi
+
+if [[ ! -e /var/lib/cassandra/data/${keyspace} ]]; then
+    header="Waiting for Cassandra"
+    let "cnt=0"
+    $namespace_prefix netstat -na | grep -q ":7199[^0-9]"
+    while [ $? -ne 0 ]; do
+        sleep 1
+        printf "${header}."
+        header=""
+        let "cnt=$cnt + 1"
+        if [ $cnt -gt 120 ]; then
+            printf "*** ERROR: Cassandra did not come online!\n"
+            exit 1
+        fi
+        netstat -na | grep -q ":7199[^0-9]"
+    done
+    let "cnt=0"
+    $namespace_prefix netstat -na | grep "LISTEN" | awk '{ print $4 }' | grep -q ":9160\$"
+    while [ $? -ne 0 ]; do
+        sleep 1
+        printf "${header}+"
+        header=""
+        let "cnt=$cnt + 1"
+        if [ $cnt -gt 120 ]; then
+            printf "*** ERROR: Cassandra did not come online!\n"
+            exit 1
+        fi
+        $namespace_prefix netstat -na | grep "LISTEN" | awk '{ print $4 }' | grep -q ":9160\$"
+    done
+
+    printf "CREATE KEYSPACE ${keyspace} WITH strategy_class = 'SimpleStrategy' AND strategy_options:replication_factor = 2;" > /tmp/$$.cqlsh.in
+
+    $namespace_prefix cqlsh -2 < /tmp/$$.cqlsh.in
+    rm -f /tmp/$$.cqlsh.in
+
+    echo "
+    USE ${keyspace};
+    CREATE TABLE call_lists (impu text PRIMARY KEY, dummy text) WITH read_repair_chance = 1.0;" >> /tmp/$$.cqlsh.in
+
+    $namespace_prefix cqlsh -2 < /tmp/$$.cqlsh.in
+    rm -f /tmp/$$.cqlsh.in
```

Comment:
Can't you just build the command and pipe it to cqlsh (like the code that you removed did)?